jasper_reports/models/report_xml.py

```diff
@@ -90,39 +85,39 @@ def write(self, values):
 class ReportXml(models.Model):
     _inherit = 'ir.actions.report'
 
-    jasper_output = fields.Selection([('html', 'HTML'), ('csv', 'CSV'),
-                                      ('xls', 'XLS'), ('rtf', 'RTF'),
-                                      ('odt', 'ODT'), ('ods', 'ODS'),
-                                      ('txt', 'Text'), ('pdf', 'PDF')],
-                                     'Jasper Output', default='pdf')
-    jasper_file_ids = fields.One2many('ir.actions.report.xml.file',
-                                      'report_id', 'Files', help='')
+    jasper_output = fields.Selection(
+        [('html', 'HTML'), ('csv', 'CSV'),
+         ('xls', 'XLS'), ('rtf', 'RTF'),
+         ('odt', 'ODT'), ('ods', 'ODS'),
+         ('txt', 'Text'), ('pdf', 'PDF')],
+        default='pdf')
+    jasper_file_ids = fields.One2many(
+        'ir.actions.report.xml.file', 'report_id', 'Files', help='')
     # To get the model name from current models in database,we add a new field
     # and it will give us model name at create and update time.
-    jasper_model_id = fields.Many2one('ir.model', 'Model', help='Select Model')
     jasper_report = fields.Boolean('Is Jasper Report?')
     report_type = fields.Selection(selection_add=[("jasper", "Jasper")])
 
-    @api.multi
     def retrieve_jasper_attachment(self, record):
         '''Retrieve an attachment for a specific record.
 
         :param record: The record owning of the attachment.
         :param attachment_name: The optional name of the attachment.
         :return: A recordset of length <=1 or None
         '''
-        if self.attachment:
-            attachment_name = safe_eval(
-                self.attachment, {'object': record, 'time': time})
-        else:
-            attachment_name = str(self.name) + '.' + self.jasper_output
-        return self.env['ir.attachment'].search([
-            ('datas_fname', '=', attachment_name),
-            ('res_model', '=', self.model),
-            ('res_id', '=', record.id)
-        ], limit=1)
-
-    @api.multi
+        attachment_obj = self.env['ir.attachment']
+        for report in self:
+            if report.attachment:
+                attachment_name = safe_eval(
+                    report.attachment, {'object': record, 'time': time})
+            else:
+                attachment_name = str(report.name) + '.' + report.jasper_output
```

Comment:
Remove else and take `attachment_name = str(report.name) + '.' + report.jasper_output` above